File: 2673_make_costs_of_paths_equal_in_a_binary_tree/solution.cpp
// LeetCode 2673 - Make Costs of Paths Equal in a Binary Tree
// https://leetcode.com/problems/make-costs-of-paths-equal-in-a-binary-tree/

#include <vector>
#include <cstdlib>

class Solution {
public:
    int minIncrements(int n, std::vector<int>& cost) {
        int ans = 0;
        for (int i = n / 2 - 1; i >= 0; i--) {
            int l = 2 * i + 1, r = 2 * i + 2;
            ans += std::abs(cost[l] - cost[r]);
            cost[i] += std::max(cost[l], cost[r]);
        }
        return ans;
    }
};

File: 2673_make_costs_of_paths_equal_in_a_binary_tree/solution.php
<?php
// LeetCode 2673 - Make Costs of Paths Equal in a Binary Tree
// https://leetcode.com/problems/make-costs-of-paths-equal-in-a-binary-tree/

class Solution {
    function minIncrements($n, $cost) {
        $ans = 0;
        for ($i = intdiv($n, 2) - 1; $i >= 0; $i--) {
            $l = 2 * $i + 1;
            $r = 2 * $i + 2;
            $ans += abs($cost[$l] - $cost[$r]);
            $cost[$i] += max($cost[$l], $cost[$r]);
        }
        return $ans;
    }
}
